es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
(Incidentally, http://mrale.ph/irhydra/2/ is a relatively-friendly tool that lets you see the compiled/optimized assembly.)